app/presenters/specialist_document_presenter.rb

```diff
@@ -0,0 +1,19 @@
+class SpecialistDocumentPresenter < ContentItemPresenter
+  def contents
+    return [] unless show_contents_list?
+
+    content_item.headers
+  end
+
+private
+
+  attr_reader :view_context
+
+  def show_contents_list?
+    content_item.headers.present? && level_two_headings?
+  end
+
+  def level_two_headings?
+    content_item.headers.any? { |header| header[:level] == 2 }
```

Comment:
Can we make any assumptions about the header levels? If the roots are always level 2, then we could just replace this with `.any?`, couldn't we? (since we're nesting the lower level ones). Or if the roots are sometimes level 1, this isn't going to find the level 2 ones if they're nested.

Reply:
The `headers` are created by parsing markdown added by editors. So I don't think we can make any assumptions about the header levels. 

The code that this logic is based on checks for the existence of ["h2" elements in the body text](https://github.com/alphagov/government-frontend/blob/main/app/presenters/content_item/contents_list.rb#L43). I've tried to do the same thing, but in a nicer way and use the `headers` object we access to.

Reply:
> roots are sometimes level 1, this isn't going to find the level 2 ones if they're nested

If there are multiple H1s there are bigger problems with the docs. But it does mean that a contents list won't be created if there are just H3s.